pkg/controller/operators/catalog/operator.go

```diff
@@ -1418,9 +1453,79 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 	ensurer := newStepEnsurer(kubeclient, crclient, dynamicClient)
 	b := newBuilder(kubeclient, dynamicClient, o.csvProvidedAPIsIndexer, o.logger)
 
+	refForStep := func(step *v1alpha1.Step) *UnpackedBundleReference {
```

Comment:
I chose not to do that originally because the builder does not exist prior to 4.5 and I knew we intend to backport this.

But this is a good point - instead of adding to the builder, I'll make a new interface / method for resolving manifests and handling the step cache. In 4.5, the builder can depend on that interface, and in < 4.4 it can be used directly in the catalog operator.